### PR TITLE
Add new option to migration generate: exportDataFromTables

### PR DESCRIPTION
**Pull request by Zaszczyk**
Hello!

* Type: new feature
* Link to issue:

**In raising this pull request, I confirm the following (please check boxes):**

- [X] I have read and understood the [Contributing Guidelines][:contrib:]
- [X] I have checked that another pull request for this purpose does not exist
- [ ] I wrote some tests for this PR

Small description of change:
I have added new option for migration generate: `exportDataFromTables`. It allows to export data from specific tables, which should be comma separated.

Thanks

[:contrib:]: https://github.com/phalcon/phalcon-devtools/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by Zaszczyk** [comment]
@CameronHall I missed `phpcs`, already fixed

File: scripts/Phalcon/Mvc/Model/Migration.php [hunk 1]
```diff
@@ -178,13 +178,13 @@ public static function setMigrationPath($path)
      *
      * @return array
      */
-    public static function generateAll(ItemInterface $version, $exportData = null)
+    public static function generateAll(ItemInterface $version, $exportData = null, $exportDataOfTables = null)
```
Comment (CameronHall) [inline scripts/Phalcon/Mvc/Model/Migration.php]: Can you please rename any instance of `exportDataOfTables` to `exportDataFromTables` to maintain consistency with `Migration.php` and `Migrations.php`.

File: scripts/Phalcon/Mvc/Model/Migration.php [hunk 2]
```diff
@@ -178,13 +178,13 @@ public static function setMigrationPath($path)
...
     {
         $classDefinition = [];
         $schema = Utils::resolveDbSchema(self::$databaseConfig);
 
         foreach (self::$connection->listTables($schema) as $table) {
-            $classDefinition[$table] = self::generate($version, $table, $exportData);
+            $classDefinition[$table] = self::generate($version, $table, $exportData, $exportDataOfTables);
```
Comment (CameronHall) [inline scripts/Phalcon/Mvc/Model/Migration.php]: Here.

File: scripts/Phalcon/Mvc/Model/Migration.php [hunk 3]
```diff
@@ -200,6 +200,11 @@ public static function getDbName()
         return self::$databaseConfig->get('dbname');
     }
 
+    public static function shouldExportDataOfTable($table, $exportDataFromTables)
```
Comment (CameronHall) [inline scripts/Phalcon/Mvc/Model/Migration.php]: Please rename to `shouldExportDataFromTable`

File: scripts/Phalcon/Mvc/Model/Migration.php [hunk 4]
```diff
@@ -399,7 +408,7 @@ public static function generate(ItemInterface $version, $table, $exportData = nu
         // up()
         $classData .= $snippet->getMigrationUp();
 
-        if ($exportData == 'always') {
+        if ($exportData == 'always' || self::shouldExportDataOfTable($table, $exportDataOfTables)) {
```
Comment (CameronHall) [inline scripts/Phalcon/Mvc/Model/Migration.php]: Please rename `exportDataOfTables` to `exportDataFromTables`

File: scripts/Phalcon/Mvc/Model/Migration.php [hunk 5]
```diff
@@ -408,22 +417,24 @@ public static function generate(ItemInterface $version, $table, $exportData = nu
         // down()
         $classData .= $snippet->getMigrationDown();
 
-        if ($exportData == 'always') {
+        if ($exportData == 'always' || self::shouldExportDataOfTable($table, $exportDataOfTables)) {
```
Comment (CameronHall) [inline scripts/Phalcon/Mvc/Model/Migration.php]: Please rename `exportDataOfTables` to `exportDataFromTables`

File: scripts/Phalcon/Mvc/Model/Migration.php [hunk 6]
```diff
@@ -408,22 +417,24 @@ public static function generate(ItemInterface $version, $table, $exportData = nu
...
             $classData .= $snippet->getMigrationBatchDelete($table);
         }
 
         $classData .= "\n    }\n";
 
         // afterCreateTable()
-        if ($exportData == 'oncreate') {
+        if ($exportData == 'oncreate' || self::shouldExportDataOfTable($table, $exportDataOfTables)) {
```
Comment (CameronHall) [inline scripts/Phalcon/Mvc/Model/Migration.php]: Please rename `exportDataOfTables` to `exportDataFromTables`

File: scripts/Phalcon/Mvc/Model/Migration.php [hunk 7]
```diff
@@ -408,22 +417,24 @@ public static function generate(ItemInterface $version, $table, $exportData = nu
...
         }
 
         // end of class
         $classData .= "\n}\n";
 
         // dump data
-        if ($exportData == 'always' || $exportData == 'oncreate') {
+        if ($exportData == 'always' ||
+            $exportData == 'oncreate' ||
+            self::shouldExportDataOfTable($table, $exportDataOfTables)) {
```
Comment (CameronHall) [inline scripts/Phalcon/Mvc/Model/Migration.php]: Please rename `exportDataOfTables` to `exportDataFromTables`

**Comment by Zaszczyk** [comment]
@CameronHall I fixed these names, my bad :) Could you just squash commits during merge? It will be easier for both of us and I allowed for that in repository's settings.

**Comment by CameronHall** [comment]
@Zaszczyk sure can. One last thing though, sorry I just realized. Can you please target the 3.4.x branch?

Thanks.

**Comment by Zaszczyk** [comment]
@CameronHall sure, done. When can I expect release of this feature?

**Comment by niden** [comment]
Thank you @Zaszczyk

**Pull request merged**